### DISCUSSION TITLE
DefaultAssay for FindClusters after RPCA integration in seurat v5 vs seurat v4

### DISCUSSION THREAD

**Discussion by silviettapar** [satijalab/seurat#9114] [Q&A] [answered]
Hello,
I'm running the seurat v5 integration workflow using the RPCA method. Once I get the integration results and I re-join the integrated layers, I would like to perform a clustering analysis. Since I noticed that the default assay is RNA (Active assay: RNA), I tried to switch to the integrated assay before running FindClusters, as I used to do in seurat v4. However, I get this error  

rpca <- FindNeighbors(rpca, reduction = "integrated.rpca", dims = 1:30,verbose=T)
DefaultAssay(rpca) <- "integrated"
rpca <- FindClusters(rpca, resolution = seq(from=0.1,to=0.7,by=0.1), verbose = T)

Error in FindClusters.Seurat(rpca, resolution = seq(from = 0.1, to = 0.7, : Provided graph.name not present in Seurat object

The code works fine if I don't change the default assay (as done in the vignette at https://satijalab.org/seurat/articles/seurat5_integration) however the columns with the cluster results in the metadata have "RNA" in their name suggesting the clustering was done on the RNA assay (even if I computed the SNN on integrated.rpca ).

Is it correct not to set DefaultAssay as "integrated" in seurat v5 workflow after getting the integration results?
Thank you!

**Comment by Alexis-Varin** [comment, marked as answer]
The integration results are not stored in an assay anymore but are stored as a reduction (here integrated.rpca), precisely as PCA corrected coordinates, from which UMAP coordinates and clusters will be calculated. This is why in Seurat v5, `RunPCA` must be run before integration, while in Seurat v4, it was run after integration, you need the PCA coordinates to calculate the integration corrected coordinates in v5.

You can always revert back to a Seurat v4 integration (by using `options(Seurat.object.assay.version = "v3")`) or on your object :
`pbmc3k[["RNA3"]] <- as(object = pbmc3k[["RNA"]], Class = "Assay")`
Before running a Seurat v4 integration pipeline (using `FindIntegationAnchors` and `IntegrateData`)

More info [here](https://satijalab.org/seurat/articles/seurat5_essential_commands)

But basically, everything is good here, your clusters have been calculated on your integrated data when you specify 'integrated.rpca', which should be the name set in the `new.reduction` parameter of `IntegrateLayers`. If you look at documentation for [FindNeighbors](https://satijalab.org/seurat/reference/findneighbors) or [RunUMAP](https://satijalab.org/seurat/reference/runumap) the `reduction` parameter is set by default to 'pca' for Seurat method, which will be the unintegrated data; if you don't change the `reduction` parameter of `FindNeighbors` and `RunUMAP` to a reduction other than 'pca', even if you run an integration pipeline, Seurat v5 will not use integrated data. I think the Seurat team hasn't communicated on how important that parameter is.

**Comment by silviettapar** [comment]
Thank you Alexis! Very clear!
